libsolidity/interface/CompilerStack.h

```diff
@@ -175,8 +175,9 @@ class CompilerStack: boost::noncopyable
 	bool analyze();
 
 	/// Parses and analyzes all source units that were added
-	/// @returns false on error.
-	bool parseAndAnalyze();
+	/// @returns false on error. If @a _astOnly is true,
+	/// we just do steps up to AST generation and tolerate parsing errors.
```

Comment:
Why not just call `parse` in that case?

Reply:
Initial AST generation is done as part of Analyze and we want an AST. We could split Analyze out into two parts, the part that does up to AST generation, and the rest. Is that preferable?

Reply:
With the bool flag, the interface is not clean.

Reply:
Ok. I will change `CompilerStack.h1 not to call `parseAndAnalyze()` but instead will do separate calls here.

Reply:
Done in 10bf0cac32490efdfb98334991b5e1cbe63d68a2 now.